tests/e2e/e2e_test.go

```diff
@@ -110,3 +114,97 @@ func (s *IntegrationTestSuite) TestSuperfluidVoting() {
 		"superfluid delegation vote overwrite not working as expected",
 	)
 }
+
+func (s *IntegrationTestSuite) TestStateSync() {
+	if s.skipStateSync {
+		s.T().Skip()
+	}
+
+	chain := s.configurer.GetChainConfig(0)
+	runningNode, err := chain.GetDefaultNode()
+	s.Require().NoError(err)
+
+	persistenrPeers := chain.GetPersistentPeers()
+
+	stateSyncHostPort := fmt.Sprintf("%s:26657", runningNode.Name)
+	stateSyncRPCServers := []string{stateSyncHostPort, stateSyncHostPort}
+
+	// get trust height and trust hash.
+	trustHeight, err := runningNode.QueryCurrentHeight()
+	s.Require().NoError(err)
+
+	trustHash, err := runningNode.QueryHashFromBlock(trustHeight)
+	s.Require().NoError(err)
+
+	stateSynchingNodeConfig := &initialization.NodeConfig{
+		Name:               "state-sync",
+		Pruning:            "default",
+		PruningKeepRecent:  "0",
+		PruningInterval:    "0",
+		SnapshotInterval:   1500,
+		SnapshotKeepRecent: 2,
+	}
+
+	tempDir, err := os.MkdirTemp("", "osmosis-e2e-statesync-")
+	s.Require().NoError(err)
+
+	// configure genesis and config files for the state-synchin node.
+	nodeInit, err := initialization.InitSingleNode(
+		chain.Id,
+		tempDir,
+		filepath.Join(runningNode.ConfigDir, "config", "genesis.json"),
+		stateSynchingNodeConfig,
+		time.Duration(chain.VotingPeriod),
+		trustHeight,
+		trustHash,
+		stateSyncRPCServers,
+		persistenrPeers,
+	)
+	s.Require().NoError(err)
+
+	stateSynchingNode := chain.CreateNode(nodeInit)
+
+	// ensure that the running node has snapshots at a height > trustHeight.
+	hasSnapshotsAvailable := func(syncInfo coretypes.SyncInfo) bool {
+		const snapshotHeight = 25
```

Comment:
Are we able to reference this number from the SnapshotInterval of the snapshot node instead?

Reply:
Good idea, addressed!